scripts/run_acceptance_tests.py

```diff
@@ -0,0 +1,214 @@
+            sys.exit(error.returncode)
+            return
+        if os.path.isdir(webpack_bundles_dir_name):
+            break
+    else:
+        # We didn't break out of the loop, meaning all attempts have failed.
+        print('Failed to complete webpack compilation, exiting...')
+        sys.exit(1)
+
+
+def install_third_party_libraries(skip_install: bool) -> None:
+    """Run the installation script.
+
+    Args:
+        skip_install: bool. Whether to skip running the installation script.
+    """
+    if not skip_install:
+        install_third_party_libs.main()
+
+
+def build_js_files(dev_mode: bool, source_maps: bool = False) -> None:
+    """Build the javascript files.
+
+    Args:
+        dev_mode: bool. Represents whether to run the related commands in dev
+            mode.
+        source_maps: bool. Represents whether to use source maps while
+            building webpack.
+    """
+    if not dev_mode:
+        print('Generating files for production mode...')
+
+        build_args = ['--prod_env']
+        if source_maps:
+            build_args.append('--source_maps')
+        build.main(args=build_args)
+
+    else:
+        build.main(args=[])
+        common.run_ng_compilation()
+        run_webpack_compilation(source_maps=source_maps)
+
+
+def run_tests(args: argparse.Namespace) -> Tuple[List[bytes], int]:
+    """Run the scripts to start end-to-end tests."""
+    if common.is_oppia_server_already_running(PORTS_USED_BY_OPPIA_PROCESSES):
+        sys.exit(1)
+
+    install_third_party_libraries(args.skip_install)
+
+    with contextlib.ExitStack() as stack:
+        dev_mode = not args.prod_env
+
+        if args.skip_build:
+            build.modify_constants(prod_env=args.prod_env)
+        else:
+            build_js_files(dev_mode, source_maps=args.source_maps)
+        stack.callback(build.set_constants_to_default)
+
+        stack.enter_context(servers.managed_redis_server())
+        stack.enter_context(servers.managed_elasticsearch_dev_server())
+        if constants.EMULATOR_MODE:
+            stack.enter_context(servers.managed_firebase_auth_emulator())
+            stack.enter_context(
+                servers.managed_cloud_datastore_emulator(clear_datastore=True))
+
+        app_yaml_path = 'app.yaml' if args.prod_env else 'app_dev.yaml'
+        stack.enter_context(servers.managed_dev_appserver(
+            app_yaml_path,
+            port=GOOGLE_APP_ENGINE_PORT,
+            log_level=args.server_log_level,
+            # Automatic restart can be disabled since we don't expect code
+            # changes to happen while the e2e tests are running.
+            automatic_restart=False,
+            skip_sdk_update_check=True,
+            env={
+                **os.environ,
+                'PORTSERVER_ADDRESS': common.PORTSERVER_SOCKET_FILEPATH,
+            }))
+
+        proc = stack.enter_context(servers.managed_acceptance_tests_server(
+                suite_name=args.suite,
```

Comment:
Indent this and the next line by 4 from the previous line, not 8.

Reply:
Done !